x-pack/libbeat/common/aws/credentials.go

```diff
@@ -44,134 +42,172 @@ type ConfigAWS struct {
+		addStaticCredentialsProviderToAwsConfig(beatsConfig, config)
+		return *config
+	}
+
+	return *config
 }
 
-func getSharedCredentialProfile(config ConfigAWS) (awssdk.Config, error) {
-	// If accessKeyID, secretAccessKey or sessionToken is not given, iam_role is not given, then load from default config
-	// Please see https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html
-	// with more details.
-	// If credential_profile_name is empty, then default profile is used.
-	logger := logp.NewLogger("getSharedCredentialProfile")
-	var options []external.Config
-	if config.ProfileName != "" {
-		options = append(options, external.WithSharedConfigProfile(config.ProfileName))
+// getConfigSharedCredentialProfile If accessKeyID, secretAccessKey or sessionToken is not given, iam_role is not given,
+// then load from default config // Please see https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html
+//	with more details. If credential_profile_name is empty, then default profile is used.
+func getConfigSharedCredentialProfile(beatsConfig ConfigAWS) (awssdk.Config, error) {
+	logger := logp.NewLogger("WithSharedConfigProfile")
+
+	var options []func(*awsConfig.LoadOptions) error
+	if beatsConfig.ProfileName != "" {
+		options = append(options, awsConfig.WithSharedConfigProfile(beatsConfig.ProfileName))
 	}
 
 	// If shared_credential_file is empty, then external.LoadDefaultAWSConfig
 	// function will load AWS config from current user's home directory.
 	// Linux/OSX: "$HOME/.aws/credentials"
 	// Windows:   "%USERPROFILE%\.aws\credentials"
-	if config.SharedCredentialFile != "" {
-		options = append(options, external.WithSharedConfigFiles([]string{config.SharedCredentialFile}))
+	if beatsConfig.SharedCredentialFile != "" {
+		options = append(options, awsConfig.WithSharedConfigFiles([]string{beatsConfig.SharedCredentialFile}))
 	}
 
-	awsConfig, err := external.LoadDefaultAWSConfig(options...)
+	cfg, err := awsConfig.LoadDefaultConfig(context.TODO(), options...)
 	if err != nil {
-		return awsConfig, fmt.Errorf("external.LoadDefaultAWSConfig failed with shared credential profile given: %w", err)
+		return cfg, fmt.Errorf("awsConfig.LoadDefaultConfig failed with shared credential profile given: [%w]", err)
+	}
+
+	// Assume IAM role if iam_role config parameter is given
+	if beatsConfig.RoleArn != "" {
+		logger.Debug("Using role arn and shared credential profile for AWS credential")
+		addStaticCredentialsProviderToAwsConfig(beatsConfig, &cfg)
+		return cfg, nil
 	}
 
 	logger.Debug("Using shared credential profile for AWS credential")
-	return awsConfig, nil
+	return cfg, nil
 }
 
-// switchToAssumeRoleProvider switches the credentials provider in the awsConfig to the `AssumeRoleProvider`.
+// switchToAssumeRoleProvider TODO (this function seems deprecated and replaced by addStaticCredentialsProviderToAwsConfig) switches the credentials provider in the awsConfig to the `AssumeRoleProvider`.
```

Comment:
I left this here as a reminder that we need to ensure what's the expectation of this function